test/test_profiler.py

```diff
@@ -1582,6 +1583,37 @@ def test_profiler_extra_cuda_copy_pattern_benchmark(self):
         shapes_factor_map = pattern.benchmark(pattern.matched_events())
         self.assertEqual(len(shapes_factor_map), 2)
 
+    def test_profiler_optimizer_single_tensor_pattern(self):
+        x = torch.ones((100, 100))
+        model = nn.Sequential(
```

Comment:
You'll want to make a new model each time since the optimizer will be coupled to the model. (Shouldn't matter, but might as well.)

Reply:
👍